src/Beacon.jsx

```diff
@@ -106,6 +98,16 @@ export class Beacon extends Component {
         this.setState({ targetElement }); // eslint-disable-line react/no-did-update-set-state
       }
     }
+    const targetElement = this.getTargetElement();
+    const targetBounds = targetElement.getBoundingClientRect();
+    const boundsChanged =
+      this.state.targetBounds.top !== targetBounds.top ||
+      this.state.targetBounds.bottom !== targetBounds.bottom ||
+      this.state.targetBounds.left !== targetBounds.left ||
+      this.state.targetBounds.right !== targetBounds.right;
+    if (boundsChanged || (this.state.targetElement !== targetElement)) {
+      this.setState({ targetElement, targetBounds }); // eslint-disable-line react/no-did-update-set-state
+    }
```

Comment:
Same deal here... turns out the bounds can change on update (e.g. when using ReactSlider).